src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -236,6 +244,13 @@ func (q *qemuAmd64) enableProtection() error {
 		q.qemuMachine.Options += "confidential-guest-support=sev"
 		logger.Info("Enabling SEV guest protection")
 		return nil
+	case snpProtection:
+		if q.qemuMachine.Options != "" {
+			q.qemuMachine.Options += ","
+		}
+		q.qemuMachine.Options += "memory-encryption=snp"
```

Comment:
I think `memory-encrpytion=` should be replaced with `confidential-guest-support=` (they left `memory-encryption` as an alias for now; see https://gitlab.com/qemu-project/qemu/-/commit/e0292d7c626 ).

Reply:
Thanks!

Reply:
I just thought about it again -- before you implement my suggetsion, make sure the `confidential-guest-support` exists in the QEMU branch that supports SNP, as it's based on an older QEMU.

Reply:
It works fine with `confidential-guest-support`, I checked and updated it